packages/server/graphql/public/mutations/batchArchiveTasks.ts

```diff
@@ -0,0 +1,48 @@
+import {getUserId} from '../../../utils/authorization'
+
+import {SubscriptionChannel} from 'parabol-client/types/constEnums'
+import publish from '../../../utils/publish'
+
+import {MutationResolvers} from '../resolverTypes'
+import isValid from '../../isValid'
+import archiveTasksForDB from '../../../safeMutations/archiveTasksForDB'
+
+const batchArchiveTasks: MutationResolvers['batchArchiveTasks'] = async (
+  _source,
+  {taskIds},
+  {authToken, dataLoader, socketId: mutatorId}
+) => {
+  const viewerId = getUserId(authToken)
+  const operationId = dataLoader.share()
+  const subOptions = {mutatorId, operationId}
+
+  // VALIDATION
+  const tasks = (await dataLoader.get('tasks').loadMany(taskIds)).filter(isValid)
+  const validTasks = tasks.filter(async ({createdBy, teamId}) => {
```

Comment:
-1 This will not filter anything because your async function technically returns a `Promise` which will be considered true.
You can check this with this small example
```ts
async function main() {
  const arr = [1,2,3,4,5,6,7,8,9,10]
  const filtered = arr.filter(async (n) => {
    console.log('filtering', n)
    return false
  })
  console.log('filtered', filtered)
}

main()
```

If you want to do async filtering, you basically can do a map to `value | null` and then `.filter(isValid)`. Something along those lines
```ts
async function main() {
  const arr = [1,2,3,4,5,6,7,8,9,10]
  const filtered = (await Promise.all(arr.map(async (n) => {
    if (n % 2 === 0) return n
    return null
  }))).filter(isValid)
  console.log('filtered', filtered)
}
```